cpp/src/slice2java/Gen.cpp

```diff
@@ -4388,17 +4397,16 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
     outi << " implements " << prxName;
     outi << sb;
 
-    // TODO: eventually remove this default constructor.
-    // Default constructor
+    // Constructor which directly takes a Reference.
     outi << sp;
-    outi << nl << "public " << prxIName << "()";
+    outi << nl << prxIName << "(com.zeroc.IceInternal.Reference ref)";
     outi << sb;
-    outi << nl << "super();";
+    outi << nl << "super(ref);";
     outi << eb;
 
     // Copy constructor
     outi << sp;
-    outi << nl << "public " << prxIName << "(com.zeroc.Ice.ObjectPrx obj)";
+    outi << nl << prxIName << "(com.zeroc.Ice.ObjectPrx obj)";
```

Comment:
Now that we're no longer using any reflection (which cares about visibility), this can be made package private.